suite-native/discovery/src/discoveryThunks.ts

```diff
@@ -1,24 +1,247 @@
+                ...discovery,
+                loaded: finishedNetworksCount,
+            },
+        });
+
+        if (finishedNetworksCount === discovery.networks.length) {
+            dispatch({
+                type: updateDiscovery.type,
+                payload: {
+                    ...discovery,
+                    status: DiscoveryStatus.COMPLETED,
+                },
+            });
+        }
+    },
+);
+
+const discoverAccountsByDescriptorThunk = createThunk(
+    `${DISCOVERY_MODULE_PREFIX}/discoverAccountsByXpubThunk`,
     async (
-        { deviceState, device }: { deviceState: string; device: TrezorDevice },
+        {
+            descriptorsBundle,
+            deviceState,
+        }: {
+            descriptorsBundle: DiscoveryDescriptorItem[];
+            deviceState: string;
+        },
         { dispatch },
     ) => {
+        let isFinalRound = false;
+
+        if (!descriptorsBundle) {
+            isFinalRound = true;
+        }
+
+        // eslint-disable-next-line no-restricted-syntax
+        for (const bundleItem of descriptorsBundle) {
+            // eslint-disable-next-line no-await-in-loop
+            const { success, payload: accountInfo } = await TrezorConnect.getAccountInfo({
+                coin: bundleItem.coin,
+                descriptor: bundleItem.descriptor,
+            });
+
+            if (success) {
+                if (accountInfo.empty) {
+                    isFinalRound = true;
+                    break;
+                }
+
+                dispatch(
+                    accountsActions.createAccount(
+                        deviceState,
+                        bundleItem,
+                        accountInfo,
+                        false,
+                        // TODO: name of newly created account will be handled in follow-up issue:
+                        // see more: https://github.com/trezor/trezor-suite/issues/9501
+                        bundleItem.descriptor,
+                    ),
+                );
+            }
+        }
+
+        return isFinalRound;
+    },
+);
+
+const discoverNetworkBatchThunk = createThunk(
+    `${DISCOVERY_MODULE_PREFIX}/discoverXpubBatchThunk`,
+    async (
+        {
+            deviceState,
+            device,
+            round = 1,
+            network,
+            mutex,
+        }: {
+            deviceState: string;
+            device: TrezorDevice;
+            round?: number;
+            network: Network;
+            mutex: DeviceAccessMutex;
+        },
+        { dispatch, getState },
+    ) => {
+        const discovery = selectDiscoveryForDevice(getState());
+
+        if (!discovery || !deviceState) {
+            return;
+        }
+
+        const accountType = network.accountType || 'normal';
+
+        const lastDiscoveredAccountIndex = (round - 1) * DISCOVERY_ROUND_SIZE;
+
+        // Skip Cardano legacy/ledger account types if device does not support it.
+        const isIncompatibleCardanoType =
```

Comment:
You mean storing all the "incompatible networks" to a const in advance? I am afraid it is not possible, because the set of networks is dependant on a connected device. So some devices might support it and some not.